### PR TITLE
Feat ride price breakdown

### PR DESCRIPTION
**Pull request by shirmagen**
https://autofleet.atlassian.net/browse/AUT-8880

### CODE REVIEW

File: examples/client/Locomotion/src/Components/BsPages/ActiveRide/index.tsx [hunk 1]
```diff
@@ -152,11 +156,7 @@ const ActiveRideContent = () => {
           <StopPointsVerticalView
             ride={ride}
           />
-          <RidePaymentDetails
-            payment={ride.payment}
-            priceAmount={ride.priceAmount}
-            priceCurrency={ride.priceCurrency}
-          />
+          <RidePaymentDetails rideId={ride.id || ''} paymentMethod={ride.payment?.paymentMethod} state={ride.state} currency={ride.priceCurrency} />
```
Comment (oriefrati) [inline examples/client/Locomotion/src/Components/BsPages/ActiveRide/index.tsx]: use rideId={ride?.id}

File: examples/client/Locomotion/src/Components/BsPages/index.tsx [hunk 2]
```diff
@@ -529,7 +529,7 @@ export const ConfirmingRide = (props: any) => {
           lottieViewStyle={{
             height: '100%', width: '100%', alignItems: 'center', justifyContent: 'center',
           }}
-          sourceProp={null}
+          sourceProp={undefined}
```
Comment (oriefrati) [inline examples/client/Locomotion/src/Components/BsPages/index.tsx:532]: why undefined and not null?
if its undefined you can remove the prop

File: examples/client/Locomotion/src/Components/InformationCard/Card.tsx [hunk 3]
```diff
@@ -38,7 +38,7 @@ const Card = ({
           </>
         )}
       </CardTitleContainer>
-      <CardText>{children}</CardText>
+      {children ? <CardText>{children}</CardText> : undefined}
```
Comment (oriefrati) [inline examples/client/Locomotion/src/Components/InformationCard/Card.tsx:41]: change undefined to null

File: examples/client/Locomotion/src/Components/InformationCard/index.tsx [hunk 4]
```diff
@@ -26,31 +26,55 @@ const InformationCard = ({
...
+            <Card
+              title={title}
+              onPress={onPress}
+              verified={verified}
+              showUnverified={showUnverified}
+              icon={icon}
+              onIconPress={onIconPress}
+            />
+          )}
       </TouchableOpacity>
```
Comment (oriefrati) [inline examples/client/Locomotion/src/Components/InformationCard/index.tsx:52]: you can set default to children, so you don't need if

File: examples/client/Locomotion/src/Components/InformationCard/index.tsx [hunk 5]
```diff
@@ -26,31 +26,55 @@ const InformationCard = ({
...
-          title={title}
-          onPress={onPress}
-          verified={verified}
-          showUnverified={showUnverified}
-          icon={icon}
-          onIconPress={onIconPress}
-        >
-          {children}
-        </Card>
+      <TouchableOpacity onPress={onPress} {...props} style={{ flex: 1 }}>
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/InformationCard/index.tsx:29]: use styled component

File: examples/client/Locomotion/src/Components/PriceCard/NoTitlePriceCard.tsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { View } from 'react-native';
+import PriceCard from '../PriceCard';
+
+const NoTitlePriceCard = (props:any) => (
+  <View style={{ marginLeft: 15 }}>
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/PriceCard/NoTitlePriceCard.tsx]: use styled component

File: examples/client/Locomotion/src/Components/RideCard/index.tsx [hunk 7]
```diff
@@ -42,27 +43,51 @@ interface RideCardProps {
...
+  }, [ride]);
+
+  return (
+    <CardContainer>
+      <DateContainer>
+        <RideDate>
+          {moment(scheduledTo).format('MMMM DD, YYYY, h:mm A')}
+        </RideDate>
+        <RideDate>
+          {ride.state === 'canceled'
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/RideCard/index.tsx]: use RIDE_STATES

File: examples/client/Locomotion/src/Components/RidePaymentDetails/index.tsx [hunk 8]
```diff
@@ -1,40 +1,91 @@
...
+    getRidePriceCalculation,
+  } = useContext(RidePageContext);
+  const updatePriceCalculation = async () => {
+    const calculation = await getRidePriceCalculation(rideId);
+    setPriceCalculation(calculation);
+  };
+
+  const getTotalAmount = () => (priceCalculation?.totalPrice || 0)
+  + (priceCalculation?.discount || 0)
+  + (priceCalculation?.additionalCharges.find(({ chargeFor }) => chargeFor === 'tip')?.amount || 0);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/RidePaymentDetails/index.tsx]: use const for tip

File: examples/client/Locomotion/src/pages/RidePriceBreakdown/index.tsx [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+  const updatePriceCalculation = async () => {
+    setLoading(true);
+    const calculation = await getRidePriceCalculation(params.rideId);
+    setPriceCalculation(calculation);
+    setLoading(false);
+  };
+
+  const updateRideFromApi = async () => {
+    setLoading(true);
+    const result = await getRideFromApi(params.rideId || ride.id || '');
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/RidePriceBreakdown/index.tsx]: dont send a request if no ride id, don't need the last fallback

File: examples/client/Locomotion/src/pages/RidePriceBreakdown/index.tsx [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+  const getSymbol = () => getCurrencySymbol(priceCalculation?.currency);
+  const getPriceWithCurrency = (amount:number) => `${getSymbol()}${amount.toFixed(2)}`;
+  const getTip = () => priceCalculation?.additionalCharges?.find(({ chargeFor }) => chargeFor === 'tip');
+  const getTotalPrice = () => getPriceWithCurrency(
+    (priceCalculation?.totalPrice || 0)
+   + (priceCalculation?.additionalCharges?.reduce((s, { amount }) => s + amount, 0) || 0)
+    + (priceCalculation?.discount || 0),
+  );
+
+  const calculationTypeToUnit = {
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/RidePriceBreakdown/index.tsx]: what is it?

File: examples/client/Locomotion/src/Components/BsPages/ActiveRide/index.tsx (hunk 1 above)
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/BsPages/ActiveRide/index.tsx]: you don't need '' fallback

**Pull request merged**